src/templates/pages/search_results.html

```diff
@@ -6,10 +6,20 @@
 {% block content %}
   <h1>{{ query_type }} Search Results</h1>
   <ul>
-    {% for game in object_list %}
-      <li>
-        <a href="{% url 'game-detail' game.id %}">{{ game.name }}</a>
-      </li>
-    {% endfor %}
+    {% if nothing_found %}
+      <p>Oops! No {{ query_type }} matching your search criteria. Please try again with different search parameters.</p>
+    {% elif query_type == "Users" %}
+      {% for profile in object_list %}
+        <li>
+          <a href="{% url 'users:user-profile' profile.user_id %}">{{ profile.display_name }}</a>
```

Comment:
I didn't catch it earlier, but there's an error in your `user_profile_detail_view` function in views.py that's affecting this line. When someone isn't logged in, they are still able to search which provides them with a list of user profiles matching the query. However, these pages aren't being made by the the detail view function because the request does not have an associated user id or pk (lines 89 and 93 in views.py). I think user profiles should still be visible even if not logged in so you could just change the `user_profile_detail_view` function to something like this `if request.user.pk == pk or not request.user.pk:` and add a log-in if statement to the associated html page to address the add friend button.

Reply:
I have fixed this issue where now clicking on a link in the search results page to view profile while not logged in displays an Add Friend button. However, when clicking on that, it redirects to Login page.